crates/lsp/src/lsp.rs

```diff
@@ -1340,4 +1348,15 @@ mod tests {
             b"Content-Length: 1235\r\nContent-Type: application/vscode-jsonrpc\r\n\r\n"
         );
     }
+
+    #[gpui::test]
+    fn test_deserialize_metals_workspace_configuration() {
+        let json = r#"{"jsonrpc":"2.0","id":"2","method":"workspace/configuration","params":{"items":[{"scopeUri":"file:///Users/mph/Devel/personal/hello-scala/","section":"metals"}]}}"#;
+        let notification = serde_json::from_str::<AnyNotification>(json);
+        assert!(
+            notification.is_ok(),
```

Comment:
Let's rewrite this test a bit: do an `expect` on the `notification` and here, assert the the parsed id is actually 2 and it's a string.

I would add a test for serialization of the integers too, as this is something we use everywhere and it would be great to prove that it works (I've checked the branch and know that it works, but still: it's hard to regress things when there's a test).

Bonus tests: if you do not mind, let's also add a similar test for non-integer-like string + integer deserialization?

Reply:
Tests updated.